### PR TITLE
Untangle WindowTreeHostFactory from ExternalWindowTreeHostFactory

### PR DESCRIPTION
**Pull request by tonikitoo**
The former is now being used by chromeos/mus, and has different API
and design if compared against the later.
    
This PR untangles both interfaces, and bringing back the original
WindowTreeHostFactory implementation, renaming our own modifications
to ExternalWindowTreeHostFactory and adapting the related c++/mojom
code accordingly.

PR also includes https://crrev.com/c/767527.
    
IMPORTANT: In a follow up work, I will split up this whole PR into
smaller fixups and merge it properly into our tree, for the sake of
easier future rebase work.
    
Issue #305

### CODE REVIEW

**Comment by msisov** [comment]
nit: s/https://crrev,com/c/767527/https://crrev.com/c/767527.  (dot instead of coma after crrev)

File: services/ui/ws/external_window_tree_host_factory.cc [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+  if (iter != properties.end())
+    metrics.bounds_in_pixels = mojo::ConvertTo<gfx::Rect>(iter->second);
+
+  iter = properties.find(ui::mojom::WindowManager::kWindowType_InitProperty);
+  if (iter != properties.end()) {
+    metrics.window_type = static_cast<ui::mojom::WindowType>(
+        mojo::ConvertTo<int32_t>(iter->second));
+  }
+
+  iter =
```
Comment (msisov) [inline services/ui/ws/external_window_tree_host_factory.cc:64]: please do git cl format  services/ui/ws/external_window_tree_host_factory.cc to align with chromium coding style

File: services/ui/ws/external_window_tree_host_factory.h [hunk 2]
```diff
@@ -0,0 +1,52 @@
+// Copyright 2016 The Chromium Authors. All rights reserved.
```
Comment (msisov) [inline services/ui/ws/external_window_tree_host_factory.h]: s/2016/2017

File: services/ui/ws/window_tree.h [hunk 3]
```diff
@@ -582,9 +582,9 @@ class WindowTree : public mojom::WindowTree,
   void GetWindowManagerClient(
       mojo::AssociatedInterfaceRequest<mojom::WindowManagerClient> internal)
       override;
-  void GetWindowTreeHostFactory(
-      mojo::AssociatedInterfaceRequest<mojom::WindowTreeHostFactory> request)
-      override;
+  void GetExternalWindowTreeHostFactory(
```
Comment (msisov) [inline services/ui/ws/window_tree.h:585]: please do git cl format on this file as well

File: services/ui/ws/external_window_tree_host_factory.cc (hunk 1 above)
Comment (tonikitoo) [reply services/ui/ws/external_window_tree_host_factory.cc:64]: this is actually the outcome of git cl format :/

File: services/ui/ws/window_tree.h (hunk 3 above)
Comment (tonikitoo) [reply services/ui/ws/window_tree.h:585]: this is actually the outcome of git cl format :/

Comment (msisov) [reply services/ui/ws/window_tree.h:585]: ok :+1:

**Review by msisov — APPROVED** [review]
L-G-T-M

**Pull request merged**